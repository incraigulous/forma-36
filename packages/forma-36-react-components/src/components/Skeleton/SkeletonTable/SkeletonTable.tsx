import React, { Component } from 'react';

import TableRow from '../../Table/TableRow';
import SkeletonTableCell from './SkeletonTableCell/SkeletonTableCell';

export type SkeletonTableProps = {
  columnCount?: number;
} & typeof defaultProps;

const defaultProps = {
  columnCount: 5,
};

export class SkeletonTable extends Component<SkeletonTableProps> {
  static defaultProps = defaultProps;

  render() {
    const { columnCount } = this.props;

    return (
      <React.Fragment>
        <TableRow>
          {Array.from(Array(columnCount)).map((_, cellIndex) => (
            <SkeletonTableCell
              key={cellIndex}
              clipId={`skeleton-cell-${cellIndex}`}
            />
          ))}
        </TableRow>
      </React.Fragment>
    );
  }
}

export default SkeletonTable;
