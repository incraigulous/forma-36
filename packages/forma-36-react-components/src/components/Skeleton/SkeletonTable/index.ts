export * from './SkeletonTable';
export { default } from './SkeletonTable';
