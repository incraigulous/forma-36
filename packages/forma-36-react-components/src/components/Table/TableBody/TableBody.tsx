import React, { Component } from 'react';

import SkeletonRow from '../../Skeleton/SkeletonRow';

export type TableBodyProps = {
  className?: string;
  style?: React.CSSProperties;
  testId?: string;
  isLoading?: boolean;
  loadingColumnCount?: number;
  children: React.ReactNode;
} & typeof defaultProps;

const defaultProps = {
  testId: 'cf-ui-table-body',
};

export class TableBody extends Component<TableBodyProps> {
  static defaultProps = defaultProps;

  render() {
    const {
      className,
      children,
      testId,
      isLoading,
      loadingColumnCount,
      ...otherProps
    } = this.props;

    return (
      <tbody data-test-id={testId} className={className} {...otherProps}>
        {isLoading ? (
          <SkeletonRow columnCount={loadingColumnCount} />
        ) : (
          children
        )}
      </tbody>
    );
  }
}

export default TableBody;
