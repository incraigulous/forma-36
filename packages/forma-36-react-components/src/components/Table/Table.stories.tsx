import React, { useState, useEffect } from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, text } from '@storybook/addon-knobs';

import notes from './Table.md';
import Table from './Table';
import TableHead from './TableHead';
import TableBody from './TableBody';
import TableCell from './TableCell';
import TableRow from './TableRow';

function DefaultStory() {
  return (
    <div style={{ width: '800px' }}>
      <Table>
        <TableHead
          isSticky={boolean('isSticky', false)}
          offsetTop={text('offsetTop', '0px')}
        >
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell>Email</TableCell>
            <TableCell>Organization role</TableCell>
            <TableCell>Last activity</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          <TableRow>
            <TableCell>Jane Roe</TableCell>
            <TableCell>jane@roe.com</TableCell>
            <TableCell>CEO</TableCell>
            <TableCell>August 29, 2018</TableCell>
          </TableRow>
          <TableRow>
            <TableCell>John Doe</TableCell>
            <TableCell>john@doe.com</TableCell>
            <TableCell>CTO</TableCell>
            <TableCell>July 27, 2019</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </div>
  );
}

function AsyncDataStory() {
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const timer = setTimeout(() => {
      setLoading(false);
    }, 3000);
    return () => clearTimeout(timer);
  }, []);

  return (
    <div style={{ width: '800px' }}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell>Email</TableCell>
            <TableCell>Organization role</TableCell>
            <TableCell>Last activity</TableCell>
          </TableRow>
        </TableHead>
        <TableBody loading={loading} numberOfRows={3} numberOfCells={4}>
          <TableRow>
            <TableCell>Jane Roe</TableCell>
            <TableCell>jane@roe.com</TableCell>
            <TableCell>CEO</TableCell>
            <TableCell>August 29, 2018</TableCell>
          </TableRow>
          <TableRow>
            <TableCell>John Doe</TableCell>
            <TableCell>john@doe.com</TableCell>
            <TableCell>CTO</TableCell>
            <TableCell>July 27, 2019</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </div>
  );
}

storiesOf('Components|Table', module)
  .addParameters({
    propTypes: [
      Table['__docgenInfo'],
      TableHead['__docgenInfo'],
      TableBody['__docgenInfo'],
      TableCell['__docgenInfo'],
      TableRow['__docgenInfo'],
    ],
  })
  .add('default', () => <DefaultStory />, { notes })
  .add('Asynchronous Data', () => <AsyncDataStory />, { notes });
