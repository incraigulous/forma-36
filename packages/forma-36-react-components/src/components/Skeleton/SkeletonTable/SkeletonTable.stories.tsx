import React from 'react';
import { storiesOf } from '@storybook/react';

import notes from './SkeletonTable.md';
import SkeletonTable from './SkeletonTable';
import Table from '../../Table';
import TableHead from '../../Table/TableHead';
import TableBody from '../../Table/TableBody';
import TableCell from '../../Table/TableCell';
import TableRow from '../../Table/TableRow';

storiesOf('Components|Skeleton/SkeletonTable', module)
  .addParameters({
    propTypes: SkeletonTable['__docgenInfo'],
  })
  .add(
    'default',
    () => (
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell>Email</TableCell>
            <TableCell>Organization role</TableCell>
            <TableCell>Last activity</TableCell>
            <TableCell>2FA status</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          <SkeletonTable />
        </TableBody>
      </Table>
    ),
    { notes },
  );
