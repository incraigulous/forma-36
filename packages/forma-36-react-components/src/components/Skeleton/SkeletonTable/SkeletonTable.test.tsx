import React from 'react';
import { shallow, mount } from 'enzyme';
import { axe } from 'jest-axe';

import SkeletonTable from './SkeletonTable';
import Table from '../../Table';
import TableBody from '../../Table/TableBody';

it('renders the component', () => {
  const output = shallow(
    <Table>
      <TableBody>
        <SkeletonTable />
      </TableBody>
    </Table>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with 10 cells', () => {
  const output = shallow(
    <Table>
      <TableBody>
        <SkeletonTable columnCount={10} />
      </TableBody>
    </Table>,
  );

  expect(output).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const output = mount(
    <Table>
      <TableBody>
        <SkeletonTable />
      </TableBody>
    </Table>,
  ).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});
