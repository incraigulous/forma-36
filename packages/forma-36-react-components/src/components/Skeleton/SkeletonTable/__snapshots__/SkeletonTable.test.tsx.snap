// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<table
  cellPadding="0"
  cellSpacing="0"
  className="Table"
  data-test-id="cf-ui-table"
>
  <TableBody
    testId="cf-ui-table-body"
  >
    <SkeletonTable
      columnCount={5}
    />
  </TableBody>
</table>
`;

exports[`renders the component with 10 cells 1`] = `
<table
  cellPadding="0"
  cellSpacing="0"
  className="Table"
  data-test-id="cf-ui-table"
>
  <TableBody
    testId="cf-ui-table-body"
  >
    <SkeletonTable
      columnCount={10}
    />
  </TableBody>
</table>
`;
